File: tritium-remote/examples/generic_mutation.rs
use simple_logger::SimpleLogger;
use std::env;

use serde_json::json;

#[tokio::main]
async fn main() {
    // logging controlled by the RUST_LOG environment variable
    SimpleLogger::new().env().init().unwrap();

    let auth_token =
        env::var("TRITIUM_AUTH_TOKEN").expect("TRITIUM_AUTH_TOKEN environment variable not set");

    let mut tritium = tritium_remote::connect(
        "ws://localhost:1234",
        &auth_token,
        Some("tritium-remote example - generic mutation".to_string()),
    )
    .await
    .expect("failed to connect");

    let document: &str = "
        mutation trigger($input:ScriptTriggerInput!) {
            manuallyTriggerScript(input: $input) {
                script {
                    status
                }
            }
        }    
    ";

    log::info!("Document: {document}");

    let variables = json!( {
        "input": {
            "action": "START",
            "path": "start_stop.py"
        }
    });
    log::info!("Variables: {variables:?}");

    let result = tritium
        .query(&document, Some(variables))
        .await
        .expect("query failed");

    log::info!("Result: {result:?}");
}

File: py-tritium-remote/python/generic_mutation.py
#!/usr/bin/env python3

import os
import asyncio
import json
import tritium_remote

SCRIPT_PATH = "start_stop.py"


async def main():
    auth_token = os.environ["TRITIUM_AUTH_TOKEN"]
    host = os.environ.get("TRITIUM_HOST", "localhost")

    print("connecting...")
    tritium = await tritium_remote.connect(
        f"ws://{host}:1234", auth_token, "Python tritium-remote example - start script"
    )

    document = """
        mutation trigger($input:ScriptTriggerInput!) {
            manuallyTriggerScript(input: $input) {
                script {
                    status
                }
            }
        }   
    """

    variables = {"input": {"action": "START", "path": "start_stop.py"}}

    response = await tritium.query(document, json.dumps(variables))
    print("Response: ", response)


asyncio.run(main())
